File: Artifact2/Original/src/taskClassPackage/TaskServiceTest.java
package taskClassPackage;

import org.junit.jupiter.api.BeforeEach;
import static org.junit.jupiter.api.Assertions.*;
import org.junit.jupiter.api.Test;

public class TaskServiceTest {

	private TaskService taskService;
	
	@BeforeEach
	public void SetUp()
	{
		taskService = new TaskService();
	}
	//This adds a task to test the system with. It creates a new task service to ensure that
	//this information is able to be accessed. It then adds in the information fed into the
	//test and then checks that the information equals what should be there.
	@Test
	public void AddTaskTest()
	{
		Task task = new Task("1", "First Task", "Task's description");
		
		assertTrue(taskService.AddTask(task));
	}
	
	//This test adds a task and deletes the information when we call upon it to do
	//so with the information found in the task_ID. We then check that the information
	//is no longer there by using that same information.
	@Test
	public void DeleteTaskTest()
	{
		
		Task task = new Task("02", "New Task", "Task description.");
		
		taskService.AddTask(task);
		
		assertTrue(taskService.DeleteTask("02"));
	}
	
	//Verifies that the task created in the delete task test was deleted.
	@Test
	public void VerifyTaskDeleted()
	{
		assertFalse(taskService.DeleteTask("02"));
	}

	
	//First this test creates a new task service and then adds information in. From there
	//the system takes the task_ID we tell it and updates the name's information to the
	//information that we want to change it to. We then have to make sure that the name
	//is what we changed it to.
	@Test
	public void UpdateTaskNameTest()
	{
		Task task = new Task("1", "First Task", "Task's description");
		
		taskService.AddTask(task);
		
		taskService.UpdateTaskName("1", "Task 1");
		
		assertEquals("Task 1", taskService.GetTask("1").GetName());
	}
	
	//This test allows us to make sure that the description update is accurate. With this
	//we have to create a new task first and add it to the list. From there we call on that
	//task_ID and provide the description we wish to change it to. After this, we check that
	//our description matches the updated one.
	public void UpdateTaskDescTest()
	{
		Task task = new Task("1", "First Task", "Task's description");
		
		taskService.AddTask(task);
		
		taskService.UpdateTaskDesc("1", "New Description");
		
		assertEquals("New Description", taskService.GetTask("1").GetDesc());
	}
	
	//Ensures that the test task is retrieved.
	@Test
	public void GetTaskTest()
	{
		Task task = new Task("01", "NewTask", "New Description");

		taskService.AddTask(task);
		
		Task taskRetrieved = taskService.GetTask("01");
		
		assertEquals(task, taskRetrieved);
	}
}

File: Artifact2/Enhanced/TaskScheduler/TaskServiceTest.cs
using Xunit;

namespace TaskScheduler

{
    //This is the unit testing class that checks that the methods located within the TaskService class work
    //as intended. There are 5 tests that go over the main components for the TaskService class.
    public class TaskServiceTest
    {
        //This is a readonly variable that accesses the TaskService class. It is used to create a new task service
        //and create a single instance. This instances helps improve efficiency of the code by setting this to
        //run automatically.
        private readonly TaskService _taskService = new();

        //This test checks that the AddTask method works. The format for the new task is to set the ID, the name,
        //and the description for a single task. It then asserts that this task was created by checking that it is
        //true based on the new task.
        [Fact]
        public void AddTaskTest()
        {
            Task task = new Task("01", "First Task", "First Task Description.");

            Assert.True(_taskService.AddTask(task));
        }

        //This test checks that the DeleteTask method does what it is meant to and removes a task that matches the ID
        //that is entered. This test creates a new task, adds it to the dictonary, runs the DeleteTask and asserts
        //that it did delete the task.
        [Fact]
        public void DeleteTaskTest()
        {
            Task task = new Task("02", "New Task", "Task Description");

            _taskService.AddTask(task);

            Assert.True(_taskService.DeleteTask(task));
        }

        //This tests that the UpdateTaskName method works. This is done by creating a new task that is the focus of
        //the update. It then adds the task to the dictionary and calls upon the task service to update the name with
        //the task that was added and what the new name is. It then asserts that the updated name has been properly
        //add to the system.
        [Fact]
        public void UpdateTaskNameTest()
        {
            Task task = new Task("03", "Third Task", "Third Task Description.");

            _taskService.AddTask(task);

            TaskService.UpdateTaskName(task, "Task 3");

            Assert.Equal("Task 3", _taskService.GetTask("03")?.Name);
        }

        //This tests that the UpdateTaskDesc method works. As with most of the tests, this creates a task that is added
        //to the dictonary for updating. Once the task is added, it calls upon the UpdateTaskDesc method that accepts
        //the new task as the one to update and what the new description is. It then asserts that the descripton was
        //updated with the new information.
        [Fact]
        public void UpdateTaskDescTest()
        {
            Task task = new Task("04", "Task 4", "Task Description.");

            _taskService.AddTask(task);

            TaskService.UpdateTaskDesc(task, "Task 4 Description.");

            Assert.Equal("Task 4 Description.", _taskService.GetTask("04")?.Description);
        }

        //This tests the GetTask method. It starts with creating a new task that is added to the dictionary. Once the
        //task is created, it ensures that it can get the Task by receiving the ID given to that task and that it does
        //equal the information it takes in.
        [Fact]
        public void GetTaskTest()
        {
            Task task = new Task("05", "Task 5", "Task Description.");

            _taskService.AddTask(task);

            Assert.Equal(task, _taskService.GetTask("05"));
        }
    }
}